src/blas/KokkosBlas3_gemm.hpp

```diff
@@ -199,15 +200,22 @@ gemm (const char transA[],
     }
   #endif // KOKKOSKERNELS_DEBUG_LEVEL > 0
 
-  // Return if degenerated matrices are provided
-  if((A.extent(0) == 0) || (A.extent(1) == 0) || (C.extent(1) == 0))
+  // Return if C matrix is degenerated
+  if(C.extent(1) == 0) {
```

Comment:
This path could also be taken if ``C.extent(0) == 0`` since C would still have no entries then. The issue was that ``A.extent(1) == 0`` doesn't imply that C has no entries (e.g. 5x0 * 0x6 -> 5x6).

Reply:
Yes, we might simply check for `((C.extent(0) == 0) || (C.extent(1) == 0))` so that if we have any dimension being empty we do not need to do anything.